src/storage/table_index_list.cpp

```diff
@@ -55,38 +57,52 @@ bool TableIndexList::NameIsUnique(const string &name) {
 	return true;
 }
 
-void TableIndexList::InitializeIndexes(ClientContext &context, DataTableInfo &table_info, bool throw_on_failure) {
-	lock_guard<mutex> lock(indexes_lock);
-	for (auto &index : indexes) {
-		if (!index->IsUnknown()) {
-			continue;
-		}
-
-		auto &unknown_index = index->Cast<UnknownIndex>();
-		auto &index_type_name = unknown_index.GetIndexType();
-
-		// Do we know the type of this index now?
-		auto index_type = context.db->config.GetIndexTypes().FindByName(index_type_name);
-		if (!index_type) {
-			if (throw_on_failure) {
-				throw MissingExtensionException(
-				    "Cannot initialize index '%s', unknown index type '%s'. You probably need to load an extension.",
-				    unknown_index.name, index_type_name);
+void TableIndexList::InitializeIndexes(ClientContext &context, DataTableInfo &table_info, const char *index_type) {
```

Comment:
Is there a reason the `index_type` is a raw pointer? Could this not just be a `const string &index_type` and we can check `.empty()` instead of checking if it's null?

I'm just worried about ownership of the memory

Reply:
I feel your concern but I opted for a pointer in order to be able to define a default value (nullptr) in the method declaration. If we had a `std::optional` we could pass an optional string const ref instead.